src/vm/functions/mod.rs

```diff
@@ -221,6 +223,25 @@ fn special_if(args: &[SymbolicExpression], env: &mut Environment, context: &Loca
     }
 }
 
+fn special_asserts(args: &[SymbolicExpression], env: &mut Environment, context: &LocalContext) -> Result<Value> {
+    check_argument_count(2, args)?;
+
+    // handle the conditional clause.
+    let conditional = eval(&args[0], env, context)?;
+
+    match conditional {
+        Value::Bool(result) => {
+            if result {
+                Ok(conditional)
+            } else {
+                let thrown = eval(&args[1], env, context)?;
+                Err(ShortReturnType::ExpectedValue(thrown.clone()).into())
```

Comment:
Should this be an `asserts!`-specific error, like `ShortReturnType::AssertionFailed`?  I thought `ShortReturnType::ExpectedValue` was specific to `expects`?

Reply:
Addressed with https://github.com/blockstack/blockstack-core/pull/1127/commits/514052911f4044cb0857677b21765ff44d1685e8